Spigot-Server-Patches/0349-Make-watchdog-print-short-dumps-every-5-seconds-the-.patch

```diff
@@ -0,0 +1,142 @@
+         this.timeoutTime = timeoutTime;
+         this.restart = restart;
++        shortTimeout = Math.min(5000, timeoutTime); // Paper - Make short timeout the lower of 5 seconds, and timeout time
+     }
+ 
+     public static void doStart(int timeoutTime, boolean restart)
+@@ -52,56 +56,73 @@ public class WatchdogThread extends Thread
+         while ( !stopping )
+         {
+             //
+-            if ( lastTick != 0 && System.currentTimeMillis() > lastTick + timeoutTime && !Boolean.getBoolean("disable.watchdog")) // Paper - Add property to disable
++            long currentTime = System.currentTimeMillis(); // Paper - do we REALLY need to call this method multiple times?
++            if ( lastTick != 0 && currentTime > lastTick + shortTimeout && !Boolean.getBoolean("disable.watchdog")) // Paper - Add property to disable and short timeout
+             {
++                // Paper start
++                boolean isLongTimeout = currentTime > lastTick + timeoutTime;
++                // Don't spam short dumps
++                if(!isLongTimeout && currentTime < lastShortDump + shortTimeout) continue;
++                lastShortDump = currentTime;
++                // Paper end
+                 Logger log = Bukkit.getServer().getLogger();
+-                log.log( Level.SEVERE, "The server has stopped responding!" );
+-                log.log( Level.SEVERE, "Please report this to https://github.com/PaperMC/Paper/issues" );
+-                log.log( Level.SEVERE, "Be sure to include ALL relevant console errors and Minecraft crash reports" );
+-                log.log( Level.SEVERE, "Paper version: " + Bukkit.getServer().getVersion() );
+-                //
+-                if(net.minecraft.server.World.haveWeSilencedAPhysicsCrash)
+-                {
+-                    log.log( Level.SEVERE, "------------------------------" );
+-                    log.log( Level.SEVERE, "During the run of the server, a physics stackoverflow was supressed" );
+-                    log.log( Level.SEVERE, "near " + net.minecraft.server.World.blockLocation);
+-                }
+-                // Paper start - Warn in watchdog if an excessive velocity was ever set
+-                if ( org.bukkit.craftbukkit.CraftServer.excessiveVelEx != null )
+-                {
+-                    log.log( Level.SEVERE, "------------------------------" );
+-                    log.log( Level.SEVERE, "During the run of the server, a plugin set an excessive velocity on an entity" );
+-                    log.log( Level.SEVERE, "This may be the cause of the issue, or it may be entirely unrelated" );
+-                    log.log( Level.SEVERE, org.bukkit.craftbukkit.CraftServer.excessiveVelEx.getMessage());
+-                    for ( StackTraceElement stack : org.bukkit.craftbukkit.CraftServer.excessiveVelEx.getStackTrace() )
++                // Paper start - Different message when it's a short timeout
++                if(isLongTimeout) {
```

Comment:
Match the style of the surrounding code, please (even if said style is ugly, which it is). Keeping consistency is more important.

Reply:
In addition, do not indent any of this code - insert the `if` statement around the code without re-formatting.